WowPacketParserModule.V2_5_1_38707/Parsers/SpellHandler.cs

```diff
@@ -0,0 +1,229 @@
+
+            packet.ReadByte("DestLocSpellCastIndex", idx);
+
+            V6_0_2_19033.Parsers.SpellHandler.ReadCreatureImmunities(packet, idx, "Immunities");
+
+            V6_0_2_19033.Parsers.SpellHandler.ReadSpellHealPrediction(packet, idx, "Predict");
+
+            packet.ResetBitReader();
+
+            var hitTargetsCount = packet.ReadBits("HitTargetsCount", 16, idx);
+            var missTargetsCount = packet.ReadBits("MissTargetsCount", 16, idx);
+            var missStatusCount = packet.ReadBits("MissStatusCount", 16, idx);
+            var remainingPowerCount = packet.ReadBits("RemainingPowerCount", 9, idx);
+
+            var hasRuneData = packet.ReadBit("HasRuneData", idx);
+            var targetPointsCount = packet.ReadBits("TargetPointsCount", 16, idx);
+            var hasAmmoDisplayId = packet.ReadBit("HasAmmoDisplayId", idx);
+            var hasAmmoInventoryType = packet.ReadBit("HasAmmoInventoryType", idx);
+
+            for (var i = 0; i < missStatusCount; ++i)
+                V6_0_2_19033.Parsers.SpellHandler.ReadSpellMissStatus(packet, idx, "MissStatus", i);
+
+            V8_0_1_27101.Parsers.SpellHandler.ReadSpellTargetData(packet, dbdata, dbdata.Spell, idx, "Target");
+
+            for (var i = 0; i < hitTargetsCount; ++i)
+                dbdata.HitTargets.Add(packet.ReadPackedGuid128("HitTarget", idx, i));
+
+            for (var i = 0; i < missTargetsCount; ++i)
+                dbdata.MissedTargets.Add(packet.ReadPackedGuid128("MissTarget", idx, i));
+
+            for (var i = 0; i < remainingPowerCount; ++i)
+                V6_0_2_19033.Parsers.SpellHandler.ReadSpellPowerData(packet, idx, "RemainingPower", i);
+
+            if (hasRuneData)
+                V7_0_3_22248.Parsers.SpellHandler.ReadRuneData(packet, idx, "RemainingRunes");
+
+            for (var i = 0; i < targetPointsCount; ++i)
+                V6_0_2_19033.Parsers.SpellHandler.ReadLocation(packet, idx, "TargetPoints", i);
+
+            if (hasAmmoDisplayId)
+                dbdata.AmmoDisplayId = packet.ReadInt32("AmmoDisplayId", idx);
+
+            if (hasAmmoInventoryType)
+                dbdata.AmmoInventoryType = (uint)packet.ReadByteE<InventoryType>("InventoryType", idx);
+        }
+
+        [Parser(Opcode.SMSG_SPELL_START)]
+        public static void HandleSpellStart(Packet packet)
+        {
+            ReadSpellCastData(packet, "Cast");
+        }
+
+        [Parser(Opcode.SMSG_SPELL_GO)]
+        public static void HandleSpellGo(Packet packet)
+        {
+            ReadSpellCastData(packet, "Cast");
+
+            packet.ResetBitReader();
+            var hasLog = packet.ReadBit();
+            if (hasLog)
+                V8_0_1_27101.Parsers.SpellHandler.ReadSpellCastLogData(packet, "LogData");
+        }
+
+        [Parser(Opcode.SMSG_SET_FLAT_SPELL_MODIFIER)]
+        [Parser(Opcode.SMSG_SET_PCT_SPELL_MODIFIER)]
+        public static void HandleSetSpellModifier(Packet packet)
```

Comment:
Not the same as `V6_0_2_19033.Parsers.HandleSetSpellModifierFlat`?

Reply:
Not the same. The 6.x handler reads the amount as a float, but in 2.5.x its a signed integer.